core/src/test/java/org/apache/iceberg/TestMetadataTableScans.java

```diff
@@ -516,6 +522,101 @@ public void testPartitionColumnNamedPartition() throws Exception {
     validateIncludesPartitionScan(tasksAndEq, 0);
   }
 
+  @Test
+  public void testPartitionsTableScanWithDeleteFilesInFilter() throws IOException {
+    Assume.assumeTrue(formatVersion == 2);
+    Configuration conf = new Configuration();
+    HadoopTables tables = new HadoopTables(conf);
+    File tableLocation = new File(temp.newFolder(), "table");
+    Table table = tables.create(
```

Comment:
On same line, I think we do not need to create a table again, as 'table' is already setup in this test.

Reply:
I use HadoopTable to load the PartitionTable, so I can read the records.